### ISSUE TITLE
Is it something wrong with grad acc?

### ISSUE THREAD

**Issue by ArlixLin** [tientrandinh/Revisiting-Reverse-Distillation#12]
I saw that:
`
            optimizer_proj.zero_grad()
            optimizer_distill.zero_grad()
            loss.backward()
            if (i + 1) % accumulation_steps == 0:
                optimizer_proj.step()
                optimizer_distill.step()
`
did you mean
`
            loss.backward()
            if (i + 1) % accumulation_steps == 0:
                optimizer_proj.step()
                optimizer_distill.step()
                optimizer_proj.zero_grad()
                optimizer_distill.zero_grad()
`

**Comment by tientrandinh** [comment]
Hi, you are right @ArlixLin,  I fixed it. Thank you
